### DISCUSSION TITLE
Vector search for datasets for few-shot prompting

### DISCUSSION THREAD

**Discussion by antoniomdk** [langfuse/langfuse#3538] [Ideas]
### Describe the feature or potential improvement

Many LLM applications benefit from few-shot prompting, and the Langfuse dataset management feature already provides a way to curate a dataset of "good examples". Being able to query those datasets via vector similarly would be really interesting, as Langfuse could serve as the CMS for the examples DB. Also, it would enable some nice AI search features in the UI for large datasets (similar to [Ultralytics Explorer](https://docs.ultralytics.com/datasets/explorer/))

Both PostgresDB and Clickhouse (for V3) support vector search, so the heavy work could be offloaded to them. I understand that opening the doors for vector search in Langfuse implies many things, so I'd just suggest scoping it down to searching dataset items, and make the users supply the embedding via API/SDK.

If you guys think this is a good idea, I can try to put a PR in place and discuss approaches.

**Comment by marcklingen** [comment]
thanks for sharing! I discussed few-shot prompting as a downstream use case for datasets with a couple of teams and this would be a great addition to better support this use case. Let's track overall interest for this here

**Comment by teenaxta** [comment]
hi @marcklingen any progress on this?
